### PR TITLE
{cae}[GCC/11.2.0] DualSPHysics v5.0.175

### PR DESCRIPTION
**Pull request by robert-mijakovic**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+homepage = 'https://dual.sphysics.org/'
+
+description = """
+DualSPHysics is based on the Smoothed Particle Hydrodynamics model named SPHysics.
+
+The code is developed to study free-surface flow phenomena where Eulerian methods
+can be difficult to apply, such as waves or impact of dam-breaks on off-shore
+structures. DualSPHysics is a set of C++, CUDA and Java codes designed to deal
+with real-life engineering problems.
+"""
```
Comment (bedroge) [inline easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb:20]:
```suggestion
description = """DualSPHysics is based on the Smoothed Particle Hydrodynamics model named SPHysics.

 The code is developed to study free-surface flow phenomena where Eulerian methods
 can be difficult to apply, such as waves or impact of dam-breaks on off-shore
 structures. DualSPHysics is a set of C++, CUDA and Java codes designed to deal
 with real-life engineering problems."""
```

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+toolchain = {'name': 'GCC', 'version': '11.2.0'}
+
+source_urls = ['https://github.com/DualSPHysics/DualSPHysics/archive/refs/tags/']
+sources = ['v%(version)s.tar.gz']
+patches = ['%(name)s-%(version)s.patch']
+checksums = [
+    'd65bb7b8b4a21582091a534142348e46cff0a288a97ef86643bc2dde15637575',
+    '24ae21134099252a597d80f25cf835dd981d0f085ce33dc801e70f1d7e045829',
```
Comment (bedroge) [inline easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb:29]:
```suggestion
    'd65bb7b8b4a21582091a534142348e46cff0a288a97ef86643bc2dde15637575',  # v5.0.175.tar.gz
    '24ae21134099252a597d80f25cf835dd981d0f085ce33dc801e70f1d7e045829',  # DualSPHysics-5.0.175.patch
```

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    ('CMake', '3.21.1'),
+]
+
+separate_build_dir = True
+
+buildopts = " && chmod +x %(builddir)s/%(name)s-%(version)s/bin/linux/*_linux64"
+
+install_cmd = """mkdir -p %(installdir)s/bin &&
+              cp -r %(builddir)s/%(name)s-%(version)s/bin/linux/* %(installdir)s/bin &&
+              cp -r %(builddir)s/%(name)s-%(version)s/src/lib/linux_gcc %(installdir)s/lib"""
```
Comment (bedroge) [inline easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb:45]: Long commands usually get split like this:
```suggestion
install_cmd = "mkdir -p %(installdir)s/bin && "
install_cmd += "cp -r %(builddir)s/%(name)s-%(version)s/bin/linux/* %(installdir)s/bin && "
install_cmd += "cp -r %(builddir)s/%(name)s-%(version)s/src/lib/linux_gcc %(installdir)s/lib"
```

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        'bin/DualSPHysics4.0_LiquidGas_linux64',
+        'bin/DualSPHysics4.0_LiquidGasCPU_linux64',
+        'bin/DualSPHysics5.0_linux64',
+        'bin/DualSPHysics5.0CPU_linux64',
+        'bin/DualSPHysics5.0_NNewtonian_linux64',
+        'bin/DualSPHysics5.0_NNewtonianCPU_linux64',
+    ],
+    'dirs': ["lib", "bin"],
+}
+
```
Comment (bedroge) [inline easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb:64]: Is is possible to add a simple `sanity_check_commands`? I had a quick look, and most/all of the binaries seem to have a `-h`, so maybe you can add the names of the binaries to a variable `local_bins`, and use that for both the sanity check paths and commands?

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        'bin/MeasureTool_linux64',
+        'bin/GenCase_MkWord_linux64',
+        'bin/DualSPHysics4.0_LiquidGas_linux64',
+        'bin/DualSPHysics4.0_LiquidGasCPU_linux64',
+        'bin/DualSPHysics5.0_linux64',
+        'bin/DualSPHysics5.0CPU_linux64',
+        'bin/DualSPHysics5.0_NNewtonian_linux64',
+        'bin/DualSPHysics5.0_NNewtonianCPU_linux64',
+    ],
+    'dirs': ["lib", "bin"],
```
Comment (bedroge) [inline easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb:62]: No need to check for `bin` here, as we're already checking for multiple files in that directory.
```suggestion
    'dirs': ["lib"],
```

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0.eb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+homepage = 'https://dual.sphysics.org/'
+
+description = """
+DualSPHysics is based on the Smoothed Particle Hydrodynamics model named SPHysics.
+
+The code is developed to study free-surface flow phenomena where Eulerian methods
+can be difficult to apply, such as waves or impact of dam-breaks on off-shore
+structures. DualSPHysics is a set of C++, CUDA and Java codes designed to deal
+with real-life engineering problems.
+"""
```
Comment (bedroge) [inline easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0.eb:19]:
```suggestion
description = """DualSPHysics is based on the Smoothed Particle Hydrodynamics model named SPHysics.

 The code is developed to study free-surface flow phenomena where Eulerian methods
 can be difficult to apply, such as waves or impact of dam-breaks on off-shore
 structures. DualSPHysics is a set of C++, CUDA and Java codes designed to deal
 with real-life engineering problems."""
```

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0.eb [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    ('CMake', '3.21.1'),
+]
+
+separate_build_dir = True
+
+buildopts = " && chmod +x %(builddir)s/%(name)s-%(version)s/bin/linux/*_linux64"
+
+install_cmd = """mkdir -p %(installdir)s/bin &&
+              cp -r %(builddir)s/%(name)s-%(version)s/bin/linux/* %(installdir)s/bin &&
+              cp -r %(builddir)s/%(name)s-%(version)s/src/lib/linux_gcc %(installdir)s/lib"""
```
Comment (bedroge) [inline easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0.eb:39]:
```suggestion
install_cmd = "mkdir -p %(installdir)s/bin && "
install_cmd += "cp -r %(builddir)s/%(name)s-%(version)s/bin/linux/* %(installdir)s/bin && "
install_cmd += "cp -r %(builddir)s/%(name)s-%(version)s/src/lib/linux_gcc %(installdir)s/lib"
```

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0.eb [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        'bin/MeasureTool_linux64',
+        'bin/GenCase_MkWord_linux64',
+        'bin/DualSPHysics4.0_LiquidGas_linux64',
+        'bin/DualSPHysics4.0_LiquidGasCPU_linux64',
+        'bin/DualSPHysics5.0_linux64',
+        'bin/DualSPHysics5.0CPU_linux64',
+        'bin/DualSPHysics5.0_NNewtonian_linux64',
+        'bin/DualSPHysics5.0_NNewtonianCPU_linux64',
+    ],
+    'dirs': ["lib", "bin"],
```
Comment (bedroge) [inline easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0.eb:56]:
```suggestion
    'dirs': ["lib"],
```

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0.eb [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+}
+
```
Comment (bedroge) [inline easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0.eb:58]: Same here w.r.t. a `sanity_check_commands`.

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175.patch [hunk 10]
```diff
@@ -0,0 +1,25 @@
+diff -Naur DualSPHysics-5.0.175/src/source/CMakeLists.txt DualSPHysics-5.0.175/src/source/CMakeLists.txt
```
Comment (bedroge) [inline easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175.patch:1]: Please add the author + short description for this patch at the top (and then update its checksum in the easyconfigs).

**Review by bedroge — CHANGES_REQUESTED** [review]

**Comment by smoors** [comment]
@robert-mijakovic friendly reminder.
do you have time to look at the suggestions made by @bedroge ?

**Comment by robert-mijakovic** [comment]
> @robert-mijakovic friendly reminder. do you have time to look at the suggestions made by @bedroge ?

Yes, sorry I was busy.
I have fixed it.

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb (hunk 4 above)
Comment (robert-mijakovic) [reply easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb:54]: Solved!

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0.eb (hunk 8 above)
Comment (smoors) [reply easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0.eb:46]: you forgot this one :)

File: easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb (hunk 5 above)
Comment (smoors) [reply easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb:52]: you forgot this one too :)

Comment (robert-mijakovic) [reply easybuild/easyconfigs/d/DualSPHysics/DualSPHysics-5.0.175-GCC-11.2.0-CUDA-11.4.1.eb]: I'm doing to many things in parallel.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node301.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/21e3aa0f63a26fab906ee6f5efbbf4c1 for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @robert-mijakovic!

**Pull request merged**